src/swaps/liquidv2.rs

```diff
@@ -355,10 +445,19 @@ impl LBtcSwapScriptV2 {
         swap_id: &str,
     ) -> Result<(OutPoint, TxOut), Error> {
         let boltz_client = BoltzApiClientV2::new(boltz_url);
-        let hex = if self.swap_type == SwapType::ReverseSubmarine {
-            boltz_client.get_reverse_tx(swap_id)?.hex
-        } else {
-            boltz_client.get_submarine_tx(swap_id)?.hex
+        let hex = match self.swap_type {
+            SwapType::Chain => {
+                boltz_client
+                    .get_chain_txs(swap_id)?
+                    .user_lock
+                    .ok_or(Error::Protocol(
+                        "No user_lock transaction for Chain Swap available".to_string(),
+                    ))?
+                    .transaction
+                    .hex
+            }
+            SwapType::ReverseSubmarine => boltz_client.get_reverse_tx(swap_id)?.hex,
+            SwapType::Submarine => boltz_client.get_submarine_tx(swap_id)?.hex,
```

Comment:
This is getting a little confusing. For submarine and chain swaps this would fetch the lockup transaction of the user. For reverse swaps the one of the server. How about returning a struct with optional `user` and `server` transaction?

Reply:
Updated the function name to `fetch_lockup_utxo_boltz` does that make it more clear?